shared/actions/chat2/index.js

```diff
@@ -763,26 +758,26 @@ const loadMoreMessages = (
   }
 
   if (action.type === Chat2Gen.loadOlderMessagesDueToScroll) {
-    paginationKey = meta.paginationKey
-    // no more to load
-    if (!paginationKey) {
-      logger.info('Load thread bail: scrolling back and no pagination key')
+    if (!state.chat2.moreToLoadMap.get(conversationIDKey)) {
+      logger.info('Load thread bail: scrolling back and at the end')
       return
     }
+    isScrollingBack = true
     numberOfMessagesToLoad = numMessagesOnScrollback
   } else {
     numberOfMessagesToLoad = numMessagesOnInitialLoad
   }
 
-  // Update bookkeeping
-  _loadingMessagesWithPaginationKey[Types.conversationIDKeyToString(conversationIDKey)] = paginationKey
-
-  // we clear on the first callback. we sometimes don't get a cached context
   let calledClear = false
   const onGotThread = function*({thread}: {thread: string}, context: 'full' | 'cached') {
     if (thread) {
       const uiMessages: RPCChatTypes.UIMessages = JSON.parse(thread)
 
+      if (!isScrollingBack && !calledClear) {
+        calledClear = true
+        yield Saga.put(Chat2Gen.createClearOrdinals({conversationIDKey}))
```

Comment:
we just always clear now

Reply:
does this make it more likely that the ordinal is the messageID for a given message (because they're matched on navigating into the thread)?